### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by Codesleuth**
Will now spit out errors in a channel specified by environment variable `HACKBOT_ERROR_CHANNEL`.
Also made the decision to make `hubot-slack` a required dependency.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -270,6 +264,10 @@ export default class Client {
     });
   }
 
+  private getAuth(emailAddress: string) {
```
Comment (jackthorley) [inline src/client.ts:267]: :+1:

File: src/index.ts [hunk 2]
```diff
@@ -35,7 +38,11 @@ import TellMeAboutMyTeamScript from './scripts/tell_me_about_my_team.script';
 import TellMeAboutTeamScript from './scripts/tell_me_about_team.script';
 
 function load(robot: RobotWithClient) {
-  robot.client = new Client(robot);
+  loadConfig(robot.logger.error.bind(robot.logger));
+
+  robot.client = new Client(Config.HACKBOT_API_URL.value, Config.HACKBOT_API_PASSWORD.value, robot);
```
Comment (jackthorley) [inline src/index.ts:43]: I love this change; centralising this and adding one place for the 'required'/validation steps makes it cleaner.

**Review by jackthorley — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (Codesleuth) [reply src/index.ts:43]: I plan to pull out the client into a package _very soon_. It has no more external dependencies :)

**Pull request merged**